redisinsight/api/src/modules/db-recommendations/providers/database-recommendations.provider.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@nestjs/common';
+import { Cluster, Command } from 'ioredis';
+import { get, max } from 'lodash';
+import { IFindRedisClientInstanceByOptions } from 'src/modules/redis/redis.service';
+import { DatabaseConnectionService } from 'src/modules/database/database-connection.service';
+import { convertRedisInfoReplyToObject } from 'src/utils';
+import { AppTool } from 'src/models';
+
+const minNumberOfCachedScripts = 10;
+
+@Injectable()
+export class RecommendationsService {
+  constructor(
+    private readonly databaseConnectionService: DatabaseConnectionService,
+  ) {}
+
+  /**
+   * Get database recommendations
+   * @param clientOptions
+   */
+  public async getRecommendations(
+    clientOptions: IFindRedisClientInstanceByOptions,
+  ) {
+    const recommendations = [];
+    if (await this.getLuaScriptRecommendation(clientOptions)) {
+      recommendations.push({ name: 'luaScript' });
+    }
+
+    return recommendations;
+  }
+
+  /**
+   * Check lua script recommendation
+   * @param clientOptions
+   */
+  async getLuaScriptRecommendation(
+    clientOptions: IFindRedisClientInstanceByOptions,
+  ): Promise<boolean> {
+    let nodes = [];
+    const client = await this.databaseConnectionService.createClient({
+      databaseId: clientOptions.instanceId,
+      namespace: AppTool.Common,
+    });
+
+    if (client instanceof Cluster) {
+      nodes = client.nodes('master');
+    } else {
+      nodes = [client];
+    }
+
+    const nodesNumbersOfCachedScripts = Promise.all(nodes.map(async (node) => {
+      const info = convertRedisInfoReplyToObject(
+        await node.sendCommand(
+          new Command('info', ['memory'], { replyEncoding: 'utf8' }),
+        ) as string,
+      );
+      return get(info, 'memory.number_of_cached_scripts', {});
+    }));
+
+    return max(await nodesNumbersOfCachedScripts) > minNumberOfCachedScripts;
```

Comment:
I don't like to do extra calls to Redis server since we already did it few seconds ago for other purpose. This is not critical but may be if you rework entire solution (see comment above) how we are calculating recommendations these unnecessary calls might be removed at all.